homeassistant/components/brunt/__init__.py

```diff
@@ -1 +1,68 @@
 """The brunt component."""
+import asyncio
+import logging
+
+from aiohttp.web_exceptions import HTTPError
+from brunt import BruntClientAsync
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+PLATFORMS = ["cover"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Component setup, run import config flow for each entry in config."""
+    if DOMAIN not in config:
+        return True
+    hass.data.setdefault(DOMAIN, {})
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+        )
+    )
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Brunt using config flow."""
+    hass.data.setdefault(DOMAIN, {})
+    session = async_get_clientsession(hass)
+    bapi = BruntClientAsync(
+        username=entry.data[CONF_USERNAME],
+        password=entry.data[CONF_PASSWORD],
+        session=session,
+    )
+    try:
+        await bapi.async_login()
+    except HTTPError as exc:
```

Comment:
Are you able to see the difference between bad credentials and a server malfunction? It seems that HTTPError will catch many situations, perhaps you could differentiate between server error and bad credentials via your `brunt ` package?

Reply:
No documentation from the API to base that on, so unfortunately hard to further tease out what went wrong, but I'll doublecheck.

Reply:
got rid of HTTPError and made it a bit more specific